model/project_ref_test.go

```diff
@@ -1822,6 +1845,28 @@ func TestGithubPermissionGroups(t *testing.T) {
 		}
 		assert.ErrorContains(p.ValidateGitHubPermissionGroups(), fmt.Sprintf("group 'second-group' for requester '%s' not found", evergreen.GithubPRRequester))
 	})
+
+	t.Run("Intersection of permissions should return most restrictive", func(t *testing.T) {
+		intersection := orgGroup[0].Intersection(orgGroup[1])
+		assert.Equal(orgGroup[0].Name, intersection.Name)
+		assert.False(intersection.NoPermissions)
+
+		assert.Equal("write", utility.FromStringPtr(intersection.Permissions.Administration), "write and admin should restrict to write")
+		assert.Equal("read", utility.FromStringPtr(intersection.Permissions.Actions), "read and write should restrict to read")
+		assert.Nil(intersection.Permissions.Contents, "nil and write should restrict to nil")
+		assert.Equal("read", utility.FromStringPtr(intersection.Permissions.Environments), "nope and read should ignore nope and restrict to read")
+		assert.Nil(intersection.Permissions.Followers, "nope1 and nope2 should be ignored and restrict to nil")
+		assert.Equal("read", utility.FromStringPtr(intersection.Permissions.Checks), "write and read should restrict to read when reversed")
+		assert.Equal("write", utility.FromStringPtr(intersection.Permissions.Metadata), "both write should restrict to write")
+
+		assert.Nil(intersection.Permissions.Emails, "an unspecified field should restrict to nil")
```

Comment:
can we please add a check to make sure that `Contents` is not included?

Reply:
It's a bit above this line (I did it alphabetical order of ones that are included and then this one which is unspecified at the end)